tabular/src/autogluon/tabular/configs/config_helper.py

```diff
@@ -0,0 +1,357 @@
+        Only valid if `refit_full` is set.
+        """
+        self.config['set_best_to_refit_full'] = set_best_to_refit_full
+        return self
+
+    def keep_only_best(self, keep_only_best=True) -> ConfigBuilder:
+        """
+        If True, only the best model and its ancestor models are saved in the outputted `predictor`. All other models are deleted.
+            If you only care about deploying the most accurate predictor with the smallest file-size and no longer need any of the other trained models or functionality beyond prediction on new data, then set: `keep_only_best=True`, `save_space=True`.
+            This is equivalent to calling `predictor.delete_models(models_to_keep='best', dry_run=False)` directly after `fit()`.
+        If used with `refit_full` and `set_best_to_refit_full`, the best model will be the refit_full model, and the original bagged best model will be deleted.
+            `refit_full` will be automatically set to 'best' in this case to avoid training models which will be later deleted.
+        """
+        self.config['keep_only_best'] = keep_only_best
+        return self
+
+    def save_space(self, save_space=True) -> ConfigBuilder:
+        """
+        If True, reduces the memory and disk size of predictor by deleting auxiliary model files that aren't needed for prediction on new data.
+            This is equivalent to calling `predictor.save_space()` directly after `fit()`.
+        This has NO impact on inference accuracy.
+        It is recommended if the only goal is to use the trained model for prediction.
+        Certain advanced functionality may no longer be available if `save_space=True`. Refer to `predictor.save_space()` documentation for more details.
+        """
+        self.config['save_space'] = save_space
+        return self
+
+    def feature_generator(self) -> FeatureGeneratorBuilder:
+        """
+        The feature generator used by AutoGluon to process the input data to the form sent to the models. This often includes automated feature generation and data cleaning.
+        It is generally recommended to keep the default feature generator unless handling an advanced use-case.
+        """
+        return FeatureGeneratorBuilder(self)
+
+    def calibrate(self, calibrate=True) -> ConfigBuilder:
+        """
+        If True and the problem_type is classification, temperature scaling will be used to calibrate the Predictor's estimated class probabilities
+        (which may improve metrics like log_loss) and will train a scalar parameter on the validation set.
+        If True and the problem_type is quantile regression, conformalization will be used to calibrate the Predictor's estimated quantiles
+        (which may improve the prediction interval coverage, and bagging could futher improve it) and will compute a set of scalar parameters on the validation set.
+        """
+        self.config['calibrate'] = calibrate
+        return self
+
+    def build(self) -> dict:
```

Comment:
So users can't specify kwargs not specified in this builder? Ex: unlabeled_data. Is this what we want?

Reply:
I don't think we should put kwargs here just because they can be easily put in the fit call directly.

```python
config = (ConfigBuilder()
          .presets('best_quality')
          .hyperparameters('light')
          .time_limit(60)
          .build())

predictor = TabularPredictor(label=label, path='/Users/ashyrkou/Projects/autogluon/temp/').fit(train_data, **config, **kwargs)
```

Both regular parameters and builders can be mixed if needed:
```python
config = (ConfigBuilder().presets('best_quality').build())
predictor = TabularPredictor(label=label, path='/Users/ashyrkou/Projects/autogluon/temp/').fit(train_data, **config, time_limit=60)
```